pkg/util/clientauthorizer/admin.go

```diff
@@ -5,23 +5,51 @@ package clientauthorizer
 
 import (
 	"crypto/tls"
-
-	"github.com/sirupsen/logrus"
+	"crypto/x509"
+	"errors"
+	"os"
 )
 
 type admin struct {
-	log *logrus.Entry
+	roots *x509.CertPool
+
+	expectedCommonName string
 }
 
-func NewAdmin(log *logrus.Entry) ClientAuthorizer {
-	return &admin{
-		log: log,
+func NewAdmin() (ClientAuthorizer, error) {
```

Comment:
We should check ADMIN_CA_CERTS and ADMIN_CLIENT_CERT_COMMON_NAME (again) in NewAdmin so we are sure it fails closed if they are invalid (e.g. empty) + unit test.